app/Repositories/Three/PostRepository.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+namespace Rogue\Repositories\Three;
+
+use Rogue\Models\Post;
+use Rogue\Services\AWS;
+use Rogue\Models\Review;
+use Rogue\Models\Signup;
+use Rogue\Services\Registrar;
+use Intervention\Image\Facades\Image;
+
+class PostRepository
+{
+    /**
+     * AWS service class instance.
+     *
+     * @var \Rogue\Services\AWS
+     */
+    protected $aws;
+
+    /**
+     * The user repository.
+     *
+     * @var \Rogue\Services\Registrar
+     */
+    protected $registrar;
+
+    /**
+     * Array of properties needed for cropping and rotating.
+     *
+     * @var array
+     */
+    protected $cropProperties = ['crop_x', 'crop_y', 'crop_width', 'crop_height', 'crop_rotate'];
+
+    /**
+     * Create a PostRepository.
+     *
+     * @param AWS $aws
+     * @param Registrar $registrar
+     */
+    public function __construct(AWS $aws, Registrar $registrar)
+    {
+        $this->aws = $aws;
+        $this->registrar = $registrar;
+    }
+
+    /**
+     * Find a post by post_id and return associated signup and tags.
+     *
+     * @param int $id
+     * @return \Rogue\Models\Post
+     */
+    public function find($id)
+    {
+        return Post::with('signup', 'tags')->findOrFail($id);
+    }
+
+    /**
+     * Create a Post.
+     *
+     * @param  array $data
+     * @param  int $signupId
+     * @return \Rogue\Models\Post|null
+     */
+    public function create(array $data, $signupId)
+    {
+        if (isset($data['file'])) {
+            // Auto-orient the photo by default based on exif data.
+            $image = Image::make($data['file'])->orientate();
+
+            $fileUrl = $this->aws->storeImage((string) $image->encode('data-url'), $signupId);
+        } else {
+            $fileUrl = 'default';
+        }
+
+        $signup = Signup::find($signupId);
+
+        // Create a post.
+        $post = new Post([
+            'signup_id' => $signup->id,
+            'northstar_id' => auth()->id(),
+            'campaign_id' => $signup->campaign_id,
+            'quantity' => $data['quantity'],
+            'url' => $fileUrl,
+            'caption' => $data['caption'],
+            'status' => isset($data['status']) ? $data['status'] : 'pending',
+            'source' => token()->client(),
+            'remote_addr' => isset($data['remote_addr']) ? $data['remote_addr'] : null,
```

Comment:
@DFurnes I saw in your comment in Shae's PR that we should use `$data->ip()` instead here. But when I do that and run tests, the test suite stops in the middle and I get this error `Segmentation fault (core dumped)`. Not sure why? 🤔

Reply:
I think he said it would be `$request->ip() ` not sure if that helps!

Reply:
Oh yup, sorry I missed this the first time around. The IP is stored on the request object, so you can't access it from the `$data` array here. You can always grab it from the helper with `request()->ip()` since I don't think we pass in a request here.

Reply:
ah great to know, thanks!